test/gtest/ucp/test_ucp_atomic.cc

```diff
@@ -124,17 +127,31 @@ class test_ucp_atomic : public test_ucp_memheap {
 
     virtual void init() {
         int atomic_mode = get_variant_value() & ATOMIC_MODE;
+        int odp         = get_variant_value() & ODP;
         const char *atomic_mode_cfg =
                         (atomic_mode == UCP_ATOMIC_MODE_CPU)    ? "cpu" :
                         (atomic_mode == UCP_ATOMIC_MODE_DEVICE) ? "device" :
                         (atomic_mode == UCP_ATOMIC_MODE_GUESS)  ? "guess" :
                         "";
         modify_config("ATOMIC_MODE", atomic_mode_cfg);
+
+        if (odp) {
+            // Do not test device atomics with ODP since SW atomic emulation
+            // is disabled with device atomics and there can be configurations
+            // without atomic lanes with ODP support which would lead to failure
+            ASSERT_NE(atomic_mode, UCP_ATOMIC_MODE_DEVICE);
```

Comment:
not sure this assert is needed. and i'd move the comment to get_test_variants

Reply:
done